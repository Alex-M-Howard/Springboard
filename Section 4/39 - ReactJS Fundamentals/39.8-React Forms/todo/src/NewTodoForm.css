label{
    margin-right: 5px;
}
input{
    margin: 2px;
}