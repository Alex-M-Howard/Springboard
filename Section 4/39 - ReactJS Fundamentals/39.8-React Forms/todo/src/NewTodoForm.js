import {React, useState} from "react";
import {v4 as uuidv4 } from "uuid";
import "./NewTodoForm.css";

const NewTodoForm = ({addTodo}) => {

    const INITIAL_STATE = {
        todo: ""
    }

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();
        addTodo({ ...formData, id: uuidv4() });
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="todo">Todo:</label>
            <input
                id="todo"
                type="text"
                name="todo"
                placeholder="New Todo"
                value={formData.todo}
                onChange={handleChange}
            ></input>
            <button>Add</button>
        </form>
    )
}

export default NewTodoForm;
