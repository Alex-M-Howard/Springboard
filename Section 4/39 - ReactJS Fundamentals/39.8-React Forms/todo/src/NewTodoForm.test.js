import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm.js";

it("should render", () => {
  render(<NewTodoForm />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
