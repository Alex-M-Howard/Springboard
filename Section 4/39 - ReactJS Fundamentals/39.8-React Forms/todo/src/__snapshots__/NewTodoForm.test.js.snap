// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo:
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="New Todo"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
